zipline/algorithm.py

```diff
@@ -1275,13 +1240,59 @@ def continuous_future(self,
         )
 
     @api_method
-    def symbols(self, *args):
+    @preprocess(
+        symbol_str=ensure_upper_case,
+        country_code=optionally(ensure_upper_case),
+    )
+    def symbol(self, symbol_str, country_code=None):
+        """Lookup an Equity by its ticker symbol.
+
+        Parameters
+        ----------
+        symbol_str : str
+            The ticker symbol for the equity to lookup.
+        country_code : str or None, optional
+            A country to limit symbol searches to.
+
+        Returns
+        -------
+        equity : Equity
+            The equity that held the ticker symbol on the current
+            symbol lookup date.
+
+        Raises
+        ------
+        SymbolNotFound
+            Raised when the symbols was not held on the current lookup date.
+
+        See Also
+        --------
+        :func:`zipline.api.set_symbol_lookup_date`
+        """
+        # If the user has not set the symbol lookup date,
+        # use the end_session as the date for symbol->sid resolution.
+        _lookup_date = self._symbol_lookup_date \
```

Comment:
Could we simplify this by initializing `self._symbol_lookup_date` to `self.sim_params.end_session` at construction time?

Reply:
I just moved this, idk if that change belongs in this pr.